File: 프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.py
from collections import Counter
def solution(topping):
    answer = 0
    total = Counter(topping)
    left = set()
    
    for i in topping:
        left.add(i)
        
        total[i] -= 1
        
        if total[i] == 0:
            del total[i]
        
        if len(left) == len(total):
            answer += 1

    return answer

File: 프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<int> topping) {
    int answer = 0;
    
    unordered_map<int,int> total;
    unordered_map<int,int> left;
    
    for (int i : topping){
        total[i]++;
    }
    
    for (int i : topping){
        left[i]++;
        total[i]--;
        
        if (total[i] == 0){
            total.erase(i);
        }
        
        if (left.size() == total.size()) {
            answer++;
        }
    }
    return answer;
}